packages/react/src/ReactBatchConfig.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {SuspenseConfig} from 'react-reconciler/src/ReactFiberSuspenseConfig';
+
+import ReactCurrentBatchConfig from './ReactCurrentBatchConfig';
+
+const DefaultSuspenseConfig: SuspenseConfig = {
+  timeoutMs: 5000,
+  loadingDelayMs: 0,
+  minLoadingDurationMs: 0,
+};
+
+// Within the scope of the callback, mark all updates as being allowed to suspend.
+export function suspendIfNeeded(scope: () => void, config?: SuspenseConfig) {
+  const previousConfig = ReactCurrentBatchConfig.suspense;
+  ReactCurrentBatchConfig.suspense = config || DefaultSuspenseConfig;
```

Comment:
Nit: Explicit comparison?

Reply:
What do we do elsewhere? I don't want any bad types like undefined or false to leak into the implementation since it might not immediately throw only deopt, and I don't want to add a bunch of prod runtime code since this should be a fairly very light operation.

Reply:
Elsewhere we check for null and undefined explicitly, like

```js
ReactCurrentBatchConfig.suspense = config !== undefined && config !== null ? config : DefaultSuspenseConfig;
```

Reply:
I guess I'm wondering why you've chosen a different pattern here, and whether we should go change the other sites. Like `root.render`: 

https://github.com/facebook/react/blob/3d8b836e22981063d0396e04eefcb10fdec600ee/packages/react-dom/src/client/ReactDOM.js#L372-L378

Reply:
Now I wonder if `null` should be resetting it to not suspend? I guess you can also pass in timeout: 0 but then that opts you out of JND.